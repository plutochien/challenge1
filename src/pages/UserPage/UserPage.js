import React from 'react';
import PropTypes from 'prop-types';
import UserCard from "../../components/UserCard/UserCard";
import UserPost from "../../components/UserPost/UserPost";

const UserPage = ({
  match: {
    params: { userId },
  },
}) => {
  return (
    <div className="container">
      <UserCard userId={userId} />
      <UserPost userId={userId} />
    </div>
  );
};

UserPage.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      userId: PropTypes.string.isRequired,
    }),
  }),
};

export default UserPage;
