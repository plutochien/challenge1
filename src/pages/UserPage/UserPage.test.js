import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import UserPage from './UserPage';
import { BrowserRouter } from 'react-router-dom';
const mockStore = configureStore([]);

describe('UserPage Component', () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      usersReducer: {
        users: [
          {
            id: 1,
            name: 'Leanne Graham',
            username: 'Bret',
            email: 'Sincere@april.biz',
            phone: '1-770-736-8031 x56442',
            website: 'hildegard.org',
            company: {
              name: 'Romaguera-Crona',
              catchPhrase: 'Multi-layered client-server neural-net',
              bs: 'harness real-time e-markets',
            },
          },
        ],
        isUserLoading: false,
      },
      postsReducer: {
        posts: [
          {
            userId: 4,
            id: 31,
            title: 'ullam ut quidem id aut vel consequuntur',
            body: 'debitis eius sed qadipisci repudiandae',
          },
        ],
        isPostLoading: false,
      },
      commentsReducer: {
        comments: [
          {
            postId: 32,
            id: 156,
            name: 'pariatur omnis in',
            email: 'Telly_Lynch@karl.co.uk',
            body: 'dolorum voluptas laborinus dolores magni',
          },
        ],
        isCommentsLoading: false,
        currentPostId: 31,
      },
    });
    store.dispatch = jest.fn();
    component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <UserPage match={{ params: { userId: 2 } }} />
        </BrowserRouter>
      </Provider>,
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
