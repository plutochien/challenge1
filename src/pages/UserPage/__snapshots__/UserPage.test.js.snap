// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserPage Component should render with given state from Redux store 1`] = `
<div
  className="container"
>
  <div
    className="sc-bdfBwQ jHdtWi"
  >
    <div
      className="sc-dlfnbm cDmebw"
    >
      <img
        alt="profile photo"
        className="sc-gsTCUz ldecOQ"
        src="https://avatars.dicebear.com/api/avataaars/2.svg"
      />
      <div
        className="sc-gKsewC buotSy"
      />
      <div
        className="sc-gKsewC fvhoIo"
      >
        @undefined
      </div>
    </div>
  </div>
  <div
    className="sc-iBPRYJ jYonTn"
  >
    <div
      className="sc-pFZIQ heJqJd"
    >
      <div
        className="sc-jrAGrp gDpZHa"
      >
        ullam ut quidem id aut vel consequuntur
      </div>
      <div
        className="sc-kEjbxe bKFDpe"
      >
        debitis eius sed qadipisci repudiandae
      </div>
      <button
        className="sc-iqHYGH wkBod"
        onClick={[Function]}
      >
        <img
          alt="expand"
          src="expand_arrow.svg"
        />
      </button>
    </div>
    <button
      className="sc-iqHYGH wkBod"
      onClick={[Function]}
    >
      <img
        alt="show all"
        src="expand_arrow.svg"
      />
       
      <span
        className="sc-fubCfw iwfPqY"
      >
        show all
      </span>
    </button>
  </div>
</div>
`;
